src/program/latex/mod.rs

```diff
@@ -0,0 +1,1600 @@
+//! LaTeX circuit generation for quil programs.
+//!
+//! - Description:
+//! Provides a feature to generate quantum circuits using the LaTeX subpackage
+//! TikZ/[`Quantikz`] for a given quil Program. This feature is callable on
+//! Program (see usage below) and returns LaTeX string which can be rendered in
+//! a LaTeX visualization tool. Be aware that not all Programs can be parsed to
+//! LaTeX. If a Program contains a gate or modifier that has not been
+//! implemented in the Supported Gates and Modifiers section below, an error
+//! message will be returned detailing whether the entire Program or which line
+//! of instruction containing the gate or modifier is unsupported.
+//!
+//! Supported Gates and Modifiers
+//!     Pauli Gates:           I, X, Y, Z
+//!     Hadamard Gate:         H
+//!     Phase Gate:            PHASE, S, T
+//!     Controlled Phase Gate: CZ, CPHASE
+//!     Controlled X Gates:    CNOT, CCNOT
+//!     New Gates:             DEFGATE
+//!     Modifiers:             CONTROLLED, DAGGER
+//!
+//! - Usage: `Program.to_latex(settings: Settings);`
+//!
+//! This module can be viewed as a self-contained partial implementation of
+//! [`Quantikz`] with all available commands listed as variants in a Command
+//! enum. View [`Quantikz`] documentation for more information.
+//!
+//! [`Quantikz`]: https://arxiv.org/pdf/1809.03842.pdf
+
+use std::collections::{BTreeMap, HashMap, HashSet};
+use std::fmt::{format, Display};
+
+use crate::expression::Expression;
+use crate::instruction::{self, Gate, Instruction, Qubit};
+use crate::Program;
+
+/// Available commands used for building circuits with the same names taken
+/// from the Quantikz documentation for easy reference. LaTeX string denoted
+/// inside `backticks`.
+///     Single wire commands: lstick, rstick, qw, meter
+///     Multi-wire commands: ctrl, targ, control, (swap, targx)
+#[derive(Debug)]
+pub enum Command {
```

Comment:
Does this need to be `pub`?

As a consumer, the most I would expect to be public from this module is the entry point to convert a `quil_rs::Program` to a (LaTeX) `String` (and thus the required `Settings`). All other types should be private or `pub(crate)`.